contracts/L1/gateway/L1Migrator.sol

```diff
@@ -97,14 +114,25 @@ contract L1Migrator is L1ArbitrumMessenger, IMigrator, EIP712 {
     bytes32 private constant MIGRATE_SENDER_TYPE_HASH =
         keccak256("MigrateSender(address l1Addr,address l2Addr)");
 
+    modifier whenFundsMigrated() {
+        require(ethMigrated && lptMigrated, "FUNDS_NOT_MIGRATED");
```

Comment:
This check will only work if there is a gap between token+lpt migration and delegate migration and someone makes sure that the Tx on L2 succeeded (is redeemed + didn't fail) within that time period

Reply:
Right good point! So, this check only ensures that by the time a user migration function is called there are tickets for LPT and ETH migration to L2 in flight, but those tickets have not necessarily been executed yet. So, if a user migration function is called and the corresponding ticket is executed on L2 *before* the other two tickets for the LPT and ETH migration then I believe the user migration ticket would revert unless someone directly sent LPT and ETH to the L2Migrator such that it has sufficient balances to execute the relevant finalization function. If the user migration ticket does revert because the L2Migrator has insufficient balances then it could still be retried on L2 if needed.

So, I think the question is whether this modifier check is useful in some way even given the above scenario. I am leaning toward no because even if the funds migration functions haven't been called yet, if someone calls the user migration functions they'll just end up with tickets reverting on L2 which is the same as the scenario above. And once the funds migration functions are called and their tickets are executed on L2 the reverting tickets can be retried. Will plan on removing this modifier check shortly unless you think there is a reason for keeping it.

Reply:
Reverted the relevant commit in 34495b3 and removed the ethMigrated and lptMigrated booleans in bd6d1dc (which has the added benefit of keeping the L1Migrator stateless).